server/tabby_server/services/library.py

```diff
@@ -0,0 +1,231 @@
+from flask import Blueprint, request
+from http import HTTPStatus
+import requests
+import os
+from dotenv import load_dotenv
+from dataclasses import dataclass
+
+"""
+Allows calling to Google Books API and returns only the attributes of the
+assembled book inside a dataclass.
+
+Does not return books that lack an ISBN 13 id.
+"""
+
+# Load the .env variables.
+load_dotenv()
+
+# Blueprint to be accessible from __main__.py.
+# To call, use: "/library/<function_route>"
+books_api = Blueprint("library", __name__)
+
+
+# Dataclass to hold book attributes from the google books api call.
+@dataclass(frozen=True, kw_only=True)
+class BookAttr:
+    isbn: int
+    title: str
+    authors: list
+    rating: str
+    summary: str
+    thumbnail: str
+    page_count: str
+    categories: list
+    publisher: str
+    published_date: str
+
+
+def parameter_handler(param="", index=-1):
+    """
+    Takes in search string parameters for google books and converts them into
+    useable strings by the Google API.
+
+    Args:
+        param: Given parameter from the caller.
+        index: Type of parameter this is supposed to fill (author, title, etc.)
+            0: title
+            1: author
+            2: publisher
+            3: subject
+            4: isbn
+            else: N/A
+
+    Returns:
+        complete_param: Completed parameter usable by Google Books API
+                        depending on type of parameter.
+    """
+
+    match (index):
+        case 0:
+            return "+intitle:" + param
+        case 1:
+            return "+inauthor:" + param
+        case 2:
+            return "+inpublisher:" + param
+        case 3:
+            return "+subject:" + param
+        case 4:
+            return "+isbn:" + param
+        case _:
+            return param
+
+
+def query_assembler(phr="", tit="", aut="", pub="", sub="", isb=""):
+    """
+    Assembles the completed query for the Google Books API.
+
+    Args:
+        phr: Phrase parameter.
+        tit: Title parameter.
+        aut: Author parameter.
+        pub: Publisher parameter.
+        sub: Subject parameter.
+        isb: ISBN parameter.
+        num: Number of passed arguments.
+    Returns:
+        complete: Fully assembled and formatted Google Books API search
+                  query.
+    """
+
+    # Assembles the query here.
+    complete = (
```

Comment:
I'm a little confused with what this part here is doing. I see that it should assemble a query here but I'm not familiar with the syntax that follows below. Can we get some comments that explains how it assembles the query using the inline if statements?

Reply:
It was just using some concatenation and ternary statements, that was all. I added some more comments still.